magicblock-committor-service/src/tasks/mod.rs

```diff
@@ -102,6 +114,116 @@ pub struct CommitTask {
     pub commit_id: u64,
     pub allow_undelegation: bool,
     pub committed_account: CommittedAccount,
+    base_account: Option<Account>,
+    force_commit_state: bool,
+}
+
+impl CommitTask {
+    // Accounts larger than COMMIT_STATE_SIZE_THRESHOLD, use CommitDiff to
+    // reduce instruction size. Below this, commit is sent as CommitState.
+    // Chose 256 as thresold seems good enough as it could hold 8 u32 fields
+    // or 4 u64 fields.
+    const COMMIT_STATE_SIZE_THRESHOLD: usize = 256;
+
+    pub async fn new<C: TaskInfoFetcher>(
+        commit_id: u64,
+        allow_undelegation: bool,
+        committed_account: CommittedAccount,
+        task_info_fetcher: &Arc<C>,
+    ) -> Self {
+        let base_account = if committed_account.account.data.len()
+            > CommitTask::COMMIT_STATE_SIZE_THRESHOLD
+        {
+            match task_info_fetcher
```

Comment:
Why don't we fetch this outside? 
We can supply base_account as an argument instead.

1. We would remove async from constructor
2. We wouldn't need TaskInfoFetcher at all in `CommitTask`
3. Task could still decide for itself if it needs to upgrade or not as was your idea

Reply:
I used `CommitTaskBuilder` which fetches the base account and then creates the task.